polars/polars-arrow/src/kernels/rolling/mod.rs

```diff
@@ -142,3 +142,9 @@ where
 pub struct RollingVarParams {
     pub ddof: u8,
 }
+
+#[derive(Clone, Copy, Debug)]
+pub struct RollingQuantileParams {
+    pub p: f64,
```

Comment:
Can you write out the names completely? What is `p`? And `interpolation`. 

We want consistent code styles and we favor complete names in the API's (internal and external).

Reply:
`p` for prob(ability) of the corresponding quantile.
`interp` is pretty common I think. There are a few other cases where `interpolation` is abbreviated in the code too so I thought it was fine. Might as well change those too yeah?

Reply:
No, please write it all out. Then everybody reading it will understand where it stands for. :)

Reply:
When you're talking about quantiles, `p` is bog-standard and unambiguous. I could go with `prob` like I did elsewhere. But I'm kind of confused about the abbreviation policy. Sometimes they're used, sometimes they're not. 

Referring to the API discussion you sent in the other thread, brevity isn't always at odds with readability or explicitness. In many cases it helps. When you use a standard, consistent, accepted shorthand, it makes it easier to read, write, and reason about. I agree that going nuts to get the absolute shortest doesn't work, but maybe it's best to define some canonical accepted abbreviations and guidelines for new ones?

Reply:
And for interpolation. `interp` is extremely common and unambiguous. I think more common than `interpol` which is used here (which always makes me laugh because I think of the police agency). You could go even further and say "interpolation" isn't enough and I should write out `interpolation_method` but at some point you're just making it longer for no reason.

Reply:
> And for interpolation. interp is extremely common and unambiguous.

Ok, but that still make it inconsentent with most of our API where we write out names completely. This is a public struct that is in our public API, so it should be written out completely. 

I am having these discussions about style with many more developers and everybody has different preferences and styles. Can we please follow the style of the host API as I think discussions about this have no winner. Nobody is right, we just have to stick to something and try to be consistent.

Reply:
I get that it makes sense to just stick to something. But it's important to settle on something that makes enough sense to everyone.
If I wasn't clear before, my point is that there are abbreviations all over the place in the public API so I have no idea what is and isn't allowed to be abbreviated. If I wrote out `interpolation` that would be inconsistent with all of the `interpol` places. I also used `probs` in the context of a quantile in several other PRs and no one had a problem.

So I'll change the abbreviations to be what is used elsewhere, that's fine for consistency. For the broader issue of abbreviations, maybe that's something worth discussing in the future (standard forms, what's ok to abbreviate, etc.)?

Reply:
>  I also used probs in the context of a quantile in several other PRs and no one had a problem.

Then I didn't notice/see it. Reviewing PR's thoroughly is harder than writing them. ;)


@stinodego probably can explain a bit more on our naming convention or if we have written that down.

Reply:
OK, well I changed `interp` to `interpol` so it's consistent and `p` to `prob`. I think that's a fair way to do it. I know everyone has to compromise a bit on style, but I do want to be a voice nudging in the direction of reasonable terseness.
The other really important reason I'm using "prob" instead of "quantile" as a parameter is that "quantile" is actually wrong. It refers to the value of the distribution, not the corresponding probability. I don't know how that made it into some libraries, but it's not good.

I think people who come from a more science background are more inclined to terseness. Julia and R use much shorter names.

> Reviewing PR's thoroughly is harder than writing them. ;)

Happy to help. I have uncompromisingly high standards for code :-)

Reply:
> there are abbreviations all over the place in the public API so I have no idea what is and isn't allowed to be abbreviated

That's a fair point and it does make it hard for developers to determine the right course of action. We are trained to conform to the existing code style.

However, this is a large code base. Our standards and API goals evolve over time, and we do not always do the best job of going back and making sure everything is consistent with those standards. And with lots of open source input from great developers such as yourself, we do not always have time to nitpick every PR. So over time, it becomes hard to determine what the "existing code style" is, exactly.

That's why I much prefer to have everyone conform to agreed-upon standards, rather than try to conform to whatever style the code base tells them. So in cases like this, while I understand the apprehension to write out names fully as it may be inconsistent with other parts of the code base, we should be vigilant and do it anyway. Then open another PR to make the other code you spotted conform to the standards as well.

Hopefully the situation will become better when we have more people working on Polars full time (I will be full-time starting next month!). That will allow us to keep higher standards and review more thoroughly.

Just my 2 cents :)

Reply:
@stinodego  Thanks for the detailed reply. I appreciate how much time and thought everyone is putting into not only making the product work well but making the code good as well

> That's why I much prefer to have everyone conform to agreed-upon standards, rather than try to conform to whatever style the code base tells them. So in cases like this, while I understand the apprehension to write out names fully as it may be inconsistent with other parts of the code base, we should be vigilant and do it anyway. Then open another PR to make the other code you spotted conform to the standards as well.

I think this is fair. It would be helpful if they were written down somewhere as unambiguously as possible. Preferably with justifications given and the occasional opportunity to comment on and improve. Like you said, things change over time. We really do share the same goals; I'm in favor of selective terseness precisely because it can make code easier to understand. If the rules haven't been codified yet, let me know how much more I have to contribute to get a say and I'll code my ass off to get there :-D

Reply:
> It would be helpful if they were written down somewhere as unambiguously as possible.

Once I figure out where, I'll write something up :)